test/index.test.js

```diff
@@ -0,0 +1,142 @@
+var test = require('tape');
+var fs = require('fs');
+var path = require('path');
+var sinon = require('sinon');
+var request = require('request');
+var stream = require('stream');
+var log = require('npmlog');
+var fse = require('fs-extra');
+var index = require('../');
+var sym = require('../lib/symlink');
+var rimraf = require('rimraf');
+
+test('setup', (assert) => {
+  if (!fs.existsSync(__dirname + '/fixtures/out')) fs.mkdirSync(__dirname + '/fixtures/out');
+  assert.end();
+});
+
+test('[install] installs a package', function(assert) {
+  var src = path.join(__dirname + '/fixtures/', 'protozero1.5.1.tar.gz');
+  var dst = path.join(__dirname + '/fixtures/out', 'protozero/1.5.1');
+  var outfile = path.join(__dirname + '/fixtures/out', 'protozero/1.5.1', 'include', 'protozero', 'byteswap.hpp');
+  var url = 'http://fakeurl.com';
+
+  var packageList = [{
+    name: 'protozero',
+    version: '1.5.1',
+    headers: true,
+    os: null,
+    awsPath: 'headers/protozero/1.5.1.tar.gz',
+    src: url,
+    dst: dst
+  }];
+
+  var buffer = fs.readFileSync(src);
+
+  var mockStream = new stream.PassThrough();
+  mockStream.push(buffer);
+  mockStream.end();
+
+  sinon.spy(mockStream, 'pipe');
+  sinon.spy(log, 'info');
+
+  sinon.stub(request, 'get').returns(mockStream);
+
+  index.install(packageList, function() {
+    sinon.assert.calledOnce(mockStream.pipe);
+    sinon.assert.calledTwice(log.info);
+    assert.equal(log.info.getCall(0).args[0], 'check');
+    assert.equal(log.info.getCall(0).args[1], 'checked for protozero (not found locally)');
+    assert.equal(log.info.getCall(1).args[0], 'tarball');
+    assert.equal(log.info.getCall(1).args[1], 'done parsing tarball for protozero');
+    assert.equal(fs.existsSync(outfile), true);
+    fse.remove(path.join(__dirname + '/fixtures/out', 'protozero'), err => {
+      if (err) return console.error(err);
+    });
+    log.info.restore();
+    request.get.restore();
+    assert.end();
+  });
+});
+
+test('[symlink] links files', function(assert) {
+  if (!fs.existsSync(__dirname + '/fixtures/out/mason_packages/.link')) fse.mkdirpSync(__dirname + '/fixtures/out/mason_packages/.link');
+
+  var appDir = process.cwd();
+  var symlinkPath = path.join(appDir, 'test/fixtures/out/mason_packages/.link');
+  
+  var paths = [
+    [appDir + '/test/fixtures/headers/protozero/1.5.1',
+      symlinkPath
+    ],
+    [appDir + '/test/fixtures/osx-x86_64/cairo/1.14.8',
+      symlinkPath
+    ]
+  ];
+  
+  sinon.spy(log, 'info');
+
+  sinon.stub(sym, 'buildLinkPaths').returns(paths);
+
+  var proto = path.join(appDir + '/test/fixtures/out', 'mason_packages/.link', 'include', 'protozero', 'byteswap.hpp');
+  var cairo = path.join(appDir + '/test/fixtures/out', 'mason_packages/.link', 'include', 'cairo', 'cairo-ft.h');
+  var masonPath = './test/fixtures/mason-versions.ini';
```

Comment:
can this fixture be named something more explicit? I imagine we'll have more than a single mason-versions.ini file over time. Best to name the fixture specific to the unit test. 

If it can't be named anything other than `mason-versions.ini` I suggest adding directories with the unit test names.

Reply:
updated to `fake-mason-versions.ini`!